### PR TITLE
libffI: refactor + add 3.3

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libffi/3.2.1**

~These are the 3.2.x changes extracted from #433.
Required for #433~

- Extracted 3.2.1-specific changes to patches
- Added 3.3

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by danimtb** [comment]
#433 was merged. Please fix conflicts in this PR

**Comment by madebr** [comment]
@danimtb This pr should have been merged first (before #433).
But I've rebased the changes to 3.2.x to this pr.

**Comment by madebr** [comment]
> 
> _config.yml_ syntax error in build 5:
> 
> ```
> Could not detect the folder changes for version 3.3 in config.yml. Verify the contents in: [folder:3.3.x]|
> ```

@uilianries @danimtb I didn't touch 3.3.x

**Comment by danimtb** [comment]
@madebr yes sorry, I mistakenly merged that one instead of this one and this is causing trouble

**Comment by danimtb** [comment]
Sorry again @madebr, I have reverted changes of the PR merge by mistake at #777 and created a "dummy" PR #779 to regenerate the packages with a recipe without hook errors.

I will merge #779 when it is ready and we could restart the changes of this PR and #433 

Thanks a lot for the help and sorry about all the mess 😕

**Comment by madebr** [comment]
@danimtb Okay! Let me know when it's safe to post again 😄

**Comment by danimtb** [comment]
@madebr #779 was merged and packages were regenerated. Now we can resume the works here!

**Comment by madebr** [comment]
I've merged the recipes for 3.2.1 and 3.3.
Some patches differ so I put the version in front of its filename.

File: recipes/libffi/all/conanfile.py [hunk 1]
```diff
@@ -20,171 +22,57 @@ class LibffiConan(ConanFile):
...
             tools.replace_in_file(configure_path, r"-install_name \$rpath/", "-install_name ")
 
-        if self.settings.compiler == "clang" and float(str(self.settings.compiler.version)) >= 7.0:
-            # https://android.googlesource.com/platform/external/libffi/+/ca22c3cb49a8cca299828c5ffad6fcfa76fdfa77
-            sysv_s_src = os.path.join(self._source_subfolder, "src", "arm", "sysv.S")
-            tools.replace_in_file(sysv_s_src, "fldmiad", "vldmia")
-            tools.replace_in_file(sysv_s_src, "fstmiad", "vstmia")
-            tools.replace_in_file(sysv_s_src, "fstmfdd\tsp!,", "vpush")
+        if Version(self.version) < "3.3":
+            if self.settings.compiler == "clang" and Version(str(self.settings.compiler.version)) >= 7.0:
```
Comment (SSE4) [inline recipes/libffi/all/conanfile.py]: maybe extract to patch as well?

Comment (danimtb) [reply recipes/libffi/all/conanfile.py]: @madebr do you want to make this change or is this ready to be merged?

Comment (madebr) [reply recipes/libffi/all/conanfile.py]: I'll do the change, but is it ok if the patch is only used in the recipe and not in the `conandata.yml`?

Comment (danimtb) [reply recipes/libffi/all/conanfile.py]: I think it better to have all patches in conadata.yml. If that makes the recipe more complex, better keep it as is. Thanks!

Comment (madebr) [reply recipes/libffi/all/conanfile.py]: Okay, ready to merge then!

**Pull request merged**